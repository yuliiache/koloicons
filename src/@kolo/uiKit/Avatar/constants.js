export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
