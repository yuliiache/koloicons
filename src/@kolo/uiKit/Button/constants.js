export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
