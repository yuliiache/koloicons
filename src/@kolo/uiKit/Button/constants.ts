export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
