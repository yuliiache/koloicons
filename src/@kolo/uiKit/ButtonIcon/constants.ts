export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
