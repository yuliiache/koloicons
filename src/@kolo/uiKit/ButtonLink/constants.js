export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

