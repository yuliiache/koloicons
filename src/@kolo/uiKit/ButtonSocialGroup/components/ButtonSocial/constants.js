export const BUTTON_SOCIAL_TEST_ID = 'buttonSocialComponent';

export const BUTTON_SOCIAL_TEXT_TEST_ID = `${BUTTON_SOCIAL_TEST_ID}Text`;
