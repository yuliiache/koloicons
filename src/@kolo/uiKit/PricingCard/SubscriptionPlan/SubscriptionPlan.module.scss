@import "@kolo/design";

.price {
  display: flex;
  align-items: center;
  margin: 16px 0 8px 0;
  gap: 8px;
}

.priceVal {
  font-weight: 500;
  font-size: 48px;
  color: $mainFontColor;
}

.priceDesc {
  max-width: 70px;
  line-height: 1.4;
  color: $secFontColor;
}
