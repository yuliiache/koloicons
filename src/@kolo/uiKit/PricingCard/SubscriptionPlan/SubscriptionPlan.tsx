import React, {FC} from 'react';

import {LabelSize, LabelTheme} from '../../Label/constants';
import Label from '../../Label/Label';
import styles from './SubscriptionPlan.module.scss';

interface SubscriptionPlanProps {
  name: string;
  price: string;
  theme: LabelTheme;
  periodInfo: string;
}
const SubscriptionPlan: FC<SubscriptionPlanProps> = ({name, price, theme, periodInfo}) => {
  return (
    <>
      <Label
        text={name}
        theme={theme}
        size={LabelSize.LARGE}
      />
      <div className={styles.price}>
        <span className={styles.priceVal}>{price}</span>
        <span className={styles.priceDesc}>{periodInfo}</span>
      </div>
    </>
  );
};

export default SubscriptionPlan;
