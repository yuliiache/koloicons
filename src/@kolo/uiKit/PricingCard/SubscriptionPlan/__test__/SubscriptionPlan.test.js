import {LabelTheme} from '@kolo/uiKit/Label/constants';
import {cleanup, render} from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';

import SubscriptionPlan from '../SubscriptionPlan';

describe('SubscriptionPlan Component', () => {
  afterEach(cleanup);

  const commonProps = {
    name: 'Personal',
    price: '$2.50',
    theme: LabelTheme.SUCCESS,
    period: 'per month, $30 billed yearly',
  };

  it('renders snapshot', () => {
    const {asFragment} = render(<SubscriptionPlan {...commonProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
