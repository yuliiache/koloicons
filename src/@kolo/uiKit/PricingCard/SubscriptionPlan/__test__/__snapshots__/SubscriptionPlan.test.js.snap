// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionPlan Component renders snapshot 1`] = `
<DocumentFragment>
  <span
    class="label success large"
    data-testid="labelComponent"
  >
    Personal
  </span>
  <div
    class="price"
  >
    <span
      class="priceVal"
    >
      $2.50
    </span>
    <span
      class="priceDesc"
    />
  </div>
</DocumentFragment>
`;
